apps/web/lib/trpc/routers/newsletter.ts

```diff
@@ -0,0 +1,44 @@
+import { env } from "@/lib/env";
+import { TRPCError } from "@trpc/server";
+import { Resend } from "resend";
+import { z } from "zod";
+import { t } from "../trpc";
+
+export const newsletterRouter = t.router({
+  signup: t.procedure
+    .input(
+      z.object({
+        email: z.string(),
```

Comment:
they got a nice utility that does some validity checking for us
```suggestion
        email: z.string().email(),
```